api/envoy/config/core/v3/protocol.proto

```diff
@@ -651,6 +651,13 @@ message Http3ProtocolOptions {
 message SchemeHeaderTransformation {
   oneof transformation {
     // Overwrite any Scheme header with the contents of this string.
+    // If set, takes precedence over match_upstream.
     string scheme_to_overwrite = 1 [(validate.rules).string = {in: "http" in: "https"}];
   }
+
+  // Set the Scheme header to match the upstream transport protocol. For example, should a
+  // request be sent to the upstream over TLS, the scheme header will be set to "https". Should the
```

Comment:
comment doesn't match code as commented below.  Do we want to set ssl over secure transports or only over TLS?

Reply:
Decided to set it only over TLS to be consistent with the downstream fallback, which also only sets it if `ssl_connection` is not null.